### PR TITLE
Add iptables4nattemplate chain element, refactor routelocalnet chain element

### PR DESCRIPTION
**Pull request by rejmond**
Signed-off-by: Sergey Shlyanin <sergey.shlyanin@xored.com>

# Description

- Add `iptables4nattemplate` chain client element that apply iptables nat rules on `Request` and restore them on `Close`. `iptables4nattemplate` is as a part of `connectioncontextkernel` client chain
- Turn `routelocalnet` to a client chain element and move it to `connectioncontextkernel` client chain
- Add `setroutelocalnet` chain server element for setting a flag to kernel mechanism

# Issue

#463
#464

### CODE REVIEW

**Review by denis-tingaikin — COMMENTED** [review]
I think you could be instrested in this https://github.com/networkservicemesh/sdk-kernel/blob/0125eaf79f2ed9aa6c6de9116f0163b9611b2829/pkg/kernel/tools/nshandle/ns_handle.go#L37-L53

File: pkg/kernel/networkservice/applyiptables4nattemplate/server.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+// WithIPTablesManager provides an option to change IPTablesManager for a chain element
+func WithIPTablesManager(iptable IPTablesManager) Option {
+	return func(server *iptablesServer) {
+		server.manager = iptable
+	}
+}
+
+// NewServer - returns a new networkservice.NetworkServiceServer that modify IPTables rules
+// by mechanism provided template on Request and rollbacks rules changes on Close
+func NewServer(options ...Option) networkservice.NetworkServiceServer {
```
Comment (denis-tingaikin) [inline pkg/kernel/networkservice/applyiptables4nattemplate/server.go:51]: Do we really need options if dont have tests for this moment?

**Comment by denis-tingaikin** [comment]
@edwarnicke , @rejmond Could we use this https://github.com/coreos/go-iptables/blob/master/iptables/iptables.go?

**Comment by edwarnicke** [comment]
@denis-tingaikin Does it simply anything?  In the background I think its still just calling iptables, right?

**Comment by denis-tingaikin** [comment]
@edwarnicke yeah, you're right. I looked quickly and dont check that. Exechelper is looking better to me here.

**Comment by edwarnicke** [comment]
@denis-tingaikin I had exactly the same thought you did and looked at this module over the weekend :) .  But sometimes you notice details I miss, thus my question :)

File: pkg/kernel/networkservice/applyiptables4nattemplate/server.go (hunk 1 above)
Comment (rejmond) [reply pkg/kernel/networkservice/applyiptables4nattemplate/server.go]: I left it for the future. I agree, it is better to remove unused feature it for now. Done.

File: pkg/kernel/networkservice/applyiptables4nattemplate/common.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package applyiptables4nattemplate
```
Comment (denis-tingaikin) [inline pkg/kernel/networkservice/applyiptables4nattemplate/common.go:20]:
```suggestion
package applyiptables
```
or
```suggestion
package iptables
```

I think could be better.

**Review by denis-tingaikin — COMMENTED** [review]
@rejmond Did we test this in real world?

**Review by denis-tingaikin — APPROVED** [review]

File: pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	if err != nil {
+		err = errors.Wrapf(err, "%s", buf.String())
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func (m *iptableManagerImpl) Apply(rulesTemplate string) error {
+	rules := strings.Split(rulesTemplate, ";")
```
Comment (denis-tingaikin) [inline pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go:61]:
```suggestion
	rules := strings.Split(rulesTemplate, "\n")
```

File: pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+	for _, rule := range rules {
+		// Remove possible tabs and spaces from a multistring template
+		rule = strings.Trim(rule, "\t ")
```
Comment (denis-tingaikin) [inline pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go:65]:
```suggestion
		rule = strings.TrimSpace(rule)
```

**Review by denis-tingaikin — COMMENTED** [review]
Added few comments others things are fine for the first step

File: pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go (hunk 3 above)
Comment (rejmond) [reply pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go]: `Apply` accepts `rules []string` now, spliting is no longer required.

File: pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go (hunk 4 above)
Comment (rejmond) [reply pkg/kernel/networkservice/connectioncontextkernel/iptables4nattemplate/common.go]: Removed Trim function at all.

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**